src/main/java/org/sasanlabs/service/vulnerability/csrf/CSRFVulnerability.java

```diff
@@ -0,0 +1,218 @@
+package org.sasanlabs.service.vulnerability.csrf;
+
+import java.io.IOException;
+import java.util.Random;
+import java.util.function.Supplier;
+import org.apache.commons.lang3.StringUtils;
+import org.sasanlabs.internal.utility.LevelConstants;
+import org.sasanlabs.internal.utility.Variant;
+import org.sasanlabs.internal.utility.annotations.AttackVector;
+import org.sasanlabs.internal.utility.annotations.VulnerableAppRequestMapping;
+import org.sasanlabs.internal.utility.annotations.VulnerableAppRestController;
+import org.sasanlabs.service.vulnerability.bean.GenericVulnerabilityResponseBean;
+import org.sasanlabs.vulnerability.types.VulnerabilityType;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+/**
+ * This class contains vulnerabilities related to CSRF.
+ *
+ * @author xxx
+ */
+@VulnerableAppRestController(descriptionLabel = "CSRF_VULNERABILITY", value = "CSRFVulnerability")
+public class CSRFVulnerability {
+
+    private static final String RTN_BADINPUT = "Bad input!";
+
+    private static final String USR_LOGIN = "id";
+    private static final String USR_PASS = "pass";
+    private static final String USR_NEWPASS = "newpass";
+
+    private static final String LOGIN = "mylogin";
+    private String passlive = "mypassword";
+    private Boolean connect = false;
+
+    private static final String RTK_PARAM = "rtkparam";
+    private static final String FIXED_TOKEN = "7877864849";
+    private String scrt = "";
+
+    private static final Random rand = new Random();
+
+    private String checkInputs(
+            String usrId, String usrPass, String usrNewPass, Boolean securityOk) {
+        Supplier<Boolean> validatorId = () -> StringUtils.isNotBlank(usrId + usrPass);
+        Supplier<Boolean> validatorNP = () -> StringUtils.isNotBlank(usrNewPass);
+        String rtn;
+        if (validatorId.get()) {
+            if (usrId.equals(LOGIN) && usrPass.equals(passlive)) {
+                connect = true;
+                rtn = "<div id=\"lgin\">[Logged in]</div> Hello " + usrId + ".";
+            } else {
+                connect = false;
+                rtn = "<div id=\"lgou\">[Not Logged in]</div> Wrong credentials.";
+            }
+        } else if (validatorNP.get()) {
+            if (connect && securityOk) {
+                passlive = usrNewPass;
+                rtn = "<div id=\"lgin\">[Logged in]</div> Password changed correctly.";
+            } else {
+                rtn =
+                        (connect
+                                ? "Password change refused."
+                                : "<div id=\"lgou\">[Not Logged in]</div> You are not logged in. Login to change password.");
```

Comment:
Hi @NMV01 ,  If we don't use cookies then I think it will not be the right way to achieve CSRF.

For example, you are setting connect flag in the server after login so even an attacker calls the same endpoint with a change password request, it will change the password for the actual user.

I think this is not the right way to implement the CSRF attack. Changing cookie should not be very complex, we already have a code that sets the cookie at (https://github.com/SasanLabs/VulnerableApp/blob/master/src/main/java/org/sasanlabs/service/vulnerability/jwt/JWTVulnerability.java#L170-L173)

Reply:
I would also suggest that once the user logs in, please hide the login form and have a logout button which have logout functionalilty and on logout, show the login form.

Reply:
Also please don't share the password between different levels so that scanner can handle it.